backend/tests/v5/unit/processors/teamspaces/teamspaces.test.js

```diff
@@ -29,9 +30,22 @@ const JobsModel = require(`${src}/models/jobs`);
 jest.mock('../../../../../src/v5/models/teamspaces');
 const TeamspacesModel = require(`${src}/models/teamspaces`);
 
+jest.mock('../../../../../src/v5/models/roles');
+const RolesModel = require(`${src}/models/roles`);
+
 jest.mock('../../../../../src/v5/utils/permissions/permissions');
 const Permissions = require(`${src}/utils/permissions/permissions`);
 
+const invalidUsername = 'invalid';
+const createTeamspaceRoleMock = RolesModel.createTeamspaceRole.mockImplementation((username) => {
```

Comment:
Instead of doing all these global mocks with increasing amount of logic, can we just use https://jestjs.io/docs/mock-function-api#mockfnmockimplementationoncefn

For situations where we want to sway from the norm.